web/lumpTypeView.ts

```diff
@@ -572,3 +572,86 @@ export function LumpTypeViewPlaypal(scaleX:number = 4, scaleY:number = 4) {
         }
     });
 }
+
+export function LumpTypeViewColormapAll(scaleX:number = 2, scaleY:number = 2) {
+    return new LumpTypeView({
+        name: "Colormap (all)",
+        icon: "assets/icons/lump-colormap.png",
+        view: (lump: WADLump, root: HTMLElement) => {
+            const cmap = lumps.WADColorMap.from(lump);
+
+            // Get PLAYPAL, or use default if it isn't found.
+            let playpal = null;
+            if (!lump.file) {
+                playpal = lumps.WADPalette.getDefault();
+            } else {
+                for (const wadlump of lump.file.lumps) {
```

Comment:
The WADFileList object has a helper method named `getPlaypal` for finding the appropriate PLAYPAL lump and returning it as a WADPalette instance. It is better to use that function than to have a reimplementation of its logic here. (Note also its `getColormap` and `getColors` methods, which are tangentially relevant.)

Currently the inspector only supports loading a single WAD; in the future it will have to support loading multiple WADs. Once it does, the WADFileList instance containing those loaded files will be passed as an argument to these `view` functions. For now, see the flat, texture, and picture views for examples of making a WADFileList instance from the inputted lump's WADFile. For example:

``` ts
export const LumpTypeViewFlatImage = new LumpTypeView({
    name: "Image",
    icon: "assets/icons/view-image.png",
    view: (lump: WADLump, root: HTMLElement) => {
        // TODO: Proper WADFileList support
        const files: WADFileList = new WADFileList([<WADFile> lump.file]);
        const flat: lumps.WADFlat = lumps.WADFlat.from(lump);
        return util.createElement({
            tag: "img",
            class: "lump-view-image",
            src: getPng64(files, flat),
            appendTo: root,
        });
    },
});
```

Note that the call to `getPng64` here will call the `getColors` method of the WADFileList, which returns an object providing both the appropriate WADPalette and WADColorMap instance for displaying the graphic.

If you use the same `files` pattern, then when supported is added a new `files: WADFileList` parameter can be added to the `view` function and the declaration removed, without having to do any other refactoring.